<a href="https://colab.research.google.com/github/JohnPaul1278/ps5/blob/main/ps5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [8]:
import time, os, sys, re
import zipfile, json, datetime, string
import numpy as np
from statistics import *

import matplotlib.pyplot as plt
import pandas as pd
import pandas_datareader as pdr
from pandas_datareader import wb
from pandas.io.formats.style import Styler

from google.colab import files

import seaborn as sns

from google.colab import data_table
data_table.enable_dataframe_formatter()
data_table.max_columns = 50

from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"

plt.style.use('classic')

In [10]:
pip install cpi==1.0.0

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 29.0/29.0 MB 10.6 MB/s eta 0:00:00


In [19]:
import cpi

/usr/local/lib/python3.10/dist-packages/cpi/__init__.py:46: StaleDataWarning: CPI data is out of date. To accurately inflate to today's dollars, you must run `cpi.update()`.
  warnings.warn(StaleDataWarning())


In [20]:
!wget -q -O eduwage.csv https://raw.githubusercontent.com/JohnPaul1278/ps5/main/eduwage.csv #I got this here: https://www.epi.org/data/#?subject=wage-education

In [21]:
eduwage= pd.read_csv('eduwage.csv')
eduwage

Date Less than HS High school Some college Bachelor's degree  \
0   2022       $16.52      $21.94       $24.81            $41.60   
1   2021       $16.74      $22.28       $24.92            $41.32   
2   2020       $17.02      $22.70       $25.44            $41.65   
3   2019       $16.11      $21.64       $24.00            $39.61   
4   2018       $15.94      $21.50       $23.70            $38.87   
5   2017       $15.92      $21.26       $23.31            $38.65   
6   2016       $15.45      $21.00       $23.35            $38.73   
7   2015       $15.25      $20.81       $23.17            $37.96   
8   2014       $14.66      $20.34       $22.48            $36.47   
9   2013       $14.62      $20.31       $22.50            $37.00   
10  2012       $14.80      $20.50       $22.56            $36.76   
11  2011       $14.96      $20.67       $22.90            $36.44   
12  2010       $15.04      $21.06       $23.56            $37.42   
13  2009       $15.41      $21.42       $23.82            $37.30   
14  2008       $15.08      $20.93       $23.52            $37.04   
15  2007       $15.19      $21.04       $23.84            $37.23   
16  2006       $14.90      $21.17       $23.69            $36.99   
17  2005       $14.96      $21.07       $23.76            $36.89   
18  2004       $15.13      $21.33       $24.11            $36.78   
19  2003       $15.27      $21.50       $24.19            $37.14   
20  2002       $15.18      $21.39       $24.25            $37.26   
21  2001       $14.88      $21.10       $24.05            $37.07   
22  2000       $14.67      $20.80       $23.68            $36.44   
23  1999       $14.61      $20.67       $23.36            $35.76   
24  1998       $14.57      $20.39       $23.02            $34.97   
25  1997       $14.07      $19.95       $22.54            $33.44   
26  1996       $13.95      $19.62       $22.11            $32.50   
27  1995       $13.95      $19.68       $22.04            $32.86   
28  1994       $14.33      $19.88       $22.26            $32.72   
29  1993       $14.60      $19.65       $22.27            $32.16   
30  1992       $14.87      $19.65       $22.26            $32.16   
31  1991       $14.99      $19.77       $22.49            $31.87   
32  1990       $15.05      $19.74       $22.58            $32.21   
33  1989       $15.46      $20.07       $22.61            $31.89   
34  1988       $15.79      $20.66       $23.03            $33.41   
35  1987       $15.76      $20.60       $23.17            $33.19   
36  1986       $16.10      $20.61       $23.12            $32.67   
37  1985       $16.08      $20.40       $22.60            $31.50   
38  1984       $16.21      $20.30       $22.37            $31.07   
39  1983       $16.43      $20.43       $22.28            $30.73   
40  1982       $16.73      $20.65       $22.43            $30.55   
41  1981       $17.07      $20.68       $22.47            $30.17   
42  1980       $17.42      $20.90       $22.73            $30.04   
43  1979       $17.95      $21.62       $23.20            $30.57   
44  1978       $17.29      $21.52       $23.25            $30.90   
45  1977       $17.59      $21.50       $22.97            $31.07   
46  1976       $17.52      $21.76       $23.49            $31.46   
47  1975       $17.30      $21.55       $23.30            $31.45   
48  1974       $17.68      $21.60       $23.32            $31.69   
49  1973       $18.06      $22.22       $24.08            $32.80   

   Advanced degree Less than HS Share High school Share Some college Share  \
0           $53.22               7.6%             26.0%              25.9%   
1           $53.45               7.5%             25.4%              26.6%   
2           $53.74               7.2%             25.2%              27.1%   
3           $51.57               8.0%             26.1%              27.7%   
4           $51.03               8.2%             25.9%              28.3%   
5           $49.40               8.4%             26.1%            

In [42]:
eduwage = eduwage.rename(columns={"Bachelor&#39;s degree": "bach",
                                  "Some college": "some",
                                  "High school": "hs",
                                  "Less than HS": "below hs",
                                  "Advanced degree": "grad",
                                  "Date": "year"})
eduwage

year below hs      hs    some Bachelor's degree    grad  \
0   2022   $16.52  $21.94  $24.81            $41.60  $53.22   
1   2021   $16.74  $22.28  $24.92            $41.32  $53.45   
2   2020   $17.02  $22.70  $25.44            $41.65  $53.74   
3   2019   $16.11  $21.64  $24.00            $39.61  $51.57   
4   2018   $15.94  $21.50  $23.70            $38.87  $51.03   
5   2017   $15.92  $21.26  $23.31            $38.65  $49.40   
6   2016   $15.45  $21.00  $23.35            $38.73  $49.64   
7   2015   $15.25  $20.81  $23.17            $37.96  $48.42   
8   2014   $14.66  $20.34  $22.48            $36.47  $47.27   
9   2013   $14.62  $20.31  $22.50            $37.00  $48.54   
10  2012   $14.80  $20.50  $22.56            $36.76  $48.47   
11  2011   $14.96  $20.67  $22.90            $36.44  $47.56   
12  2010   $15.04  $21.06  $23.56            $37.42  $48.80   
13  2009   $15.41  $21.42  $23.82            $37.30  $49.12   
14  2008   $15.08  $20.93  $23.52            $37.04  $47.55   
15  2007   $15.19  $21.04  $23.84            $37.23  $47.31   
16  2006   $14.90  $21.17  $23.69            $36.99  $47.23   
17  2005   $14.96  $21.07  $23.76            $36.89  $47.09   
18  2004   $15.13  $21.33  $24.11            $36.78  $47.48   
19  2003   $15.27  $21.50  $24.19            $37.14  $46.79   
20  2002   $15.18  $21.39  $24.25            $37.26  $47.36   
21  2001   $14.88  $21.10  $24.05            $37.07  $46.23   
22  2000   $14.67  $20.80  $23.68            $36.44  $45.99   
23  1999   $14.61  $20.67  $23.36            $35.76  $46.02   
24  1998   $14.57  $20.39  $23.02            $34.97  $44.49   
25  1997   $14.07  $19.95  $22.54            $33.44  $43.68   
26  1996   $13.95  $19.62  $22.11            $32.50  $43.35   
27  1995   $13.95  $19.68  $22.04            $32.86  $43.27   
28  1994   $14.33  $19.88  $22.26            $32.72  $43.46   
29  1993   $14.60  $19.65  $22.27            $32.16  $41.45   
30  1992   $14.87  $19.65  $22.26            $32.16  $40.94   
31  1991   $14.99  $19.77  $22.49            $31.87  $40.71   
32  1990   $15.05  $19.74  $22.58            $32.21  $39.95   
33  1989   $15.46  $20.07  $22.61            $31.89  $39.80   
34  1988   $15.79  $20.66  $23.03            $33.41  $41.12   
35  1987   $15.76  $20.60  $23.17            $33.19  $40.55   
36  1986   $16.10  $20.61  $23.12            $32.67  $39.88   
37  1985   $16.08  $20.40  $22.60            $31.50  $38.31   
38  1984   $16.21  $20.30  $22.37            $31.07  $37.45   
39  1983   $16.43  $20.43  $22.28            $30.73  $36.78   
40  1982   $16.73  $20.65  $22.43            $30.55  $36.08   
41  1981   $17.07  $20.68  $22.47            $30.17  $35.32   
42  1980   $17.42  $20.90  $22.73            $30.04  $35.53   
43  1979   $17.95  $21.62  $23.20            $30.57  $36.19   
44  1978   $17.29  $21.52  $23.25            $30.90  $37.56   
45  1977   $17.59  $21.50  $22.97            $31.07  $37.36   
46  1976   $17.52  $21.76  $23.49            $31.46  $37.50   
47  1975   $17.30  $21.55  $23.30            $31.45  $38.41   
48  1974   $17.68  $21.60  $23.32            $31.69  $38.37   
49  1973   $18.06  $22.22  $24.08            $32.80  $38.16   

   Less than HS Share High school Share Some college Share  \
0                7.6%             26.0%              25.9%   
1                7.5%             25.4%              26.6%   
2                7.2%             25.2%              27.1%   
3                8.0%             26.1%              27.7%   
4                8.2%             25.9%              28.3%   
5                8.4%             26.1%              28.7%   
6                8.6%             26.3%              29.2%   
7                8.8%             26.5%              29.2%   
8                8.7%             27.1%              29.6%   
9                8.8%             26.9%              30.0%   
10               9.1%             27.3%              30.0%   
11               9.4%             27.9%       

In [49]:
eduwage['year'] = eduwage['year'].astype(str)

In [51]:
eduwage.dtypes

year                       object
below hs                   object
hs                         object
some                       object
Bachelor's degree          object
grad                       object
Less than HS Share         object
High school Share          object
Some college Share         object
Bachelor's degree Share    object
Advanced degree Share      object
dtype: object

Just to get an idea for how the tool works, let's adjust the average hourly pay for a worker with an advanced degree in the year 2000. Will it match the average hourly pay for workers with advanced degrees today? Will it be higher?

In [31]:
cpi.inflate(45.99, 2000)

72.36881707317075

Looks like the hourly rate of forty five ninety nine is worth seventy two dollars and thirty seven cents in today's money.
That's significantly lower than the fifty three dollar and twenty two cent hourly rate that the average worker with an advanced degree earns today. The value of an hour's work has essentially shrunk by nineteen dollars!

The value of an advanced degree is measurably smaller today than it once was.

Now let's adjust the very first minimum wage from 1938 to today's standards. The first minimum wage was only $0.25! How much would that be worth today?


In [33]:
cpi.inflate(0.25, 1938)

4.804432624113476

The first minimum wage implemented in the United States would be worth about 4.80 in today's money. Our current minimum wage, which was set in 2009, is 7.25. Our economy seems to have improved at least a little in the last 85 years.

There are many more ways to use this data, however, and I want to experiment with them more. Here is where I found the cpi catalogue of items: https://github.com/palewire/cpi/blob/master/docs/items.csv

In [52]:
eduwage['hs adjusted'] = eduwage.apply(lambda x: cpi.inflate(x.hs, x.year), axis=1)

TypeError: ignored